commit 5426e2d5d5fbefb794b5a87ce390a90b449e60d0 (refs/changes/63/4163/5)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2018-06-07T18:00:24-07:00 (1 year, 4 months ago)
    
    TRITON-479 CNAPI does not allow setting of `overprovision_ratios` as a result of request id not being passed to server.sendTaskRequest()

diff --git a/lib/endpoints/servers.js b/lib/endpoints/servers.js
index 26f5070..8d3caa6 100644
--- a/lib/endpoints/servers.js
+++ b/lib/endpoints/servers.js
@@ -499,6 +499,7 @@ Server.update = function handlerServerUpdate(req, res, next) {
                     synccb: function (err, result) {
                         cb(err, result);
                     },
+                    req_id: req.getId(),
                     params: { value: req.params.overprovision_ratios  }
                 };
 
diff --git a/lib/models/server.js b/lib/models/server.js
index 76377cc..b684bfd 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -2111,6 +2111,8 @@ function (opts) {
         req_id = opts.req.getId();
     }
 
+    assert.string(req_id, 'opts.req_id');
+
     var taskstatus = {
         id: common.genId(),
         req_id: opts.req_id,
diff --git a/package.json b/package.json
index ce250f5..6438f03 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.9.0",
+  "version": "1.9.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
